feat

brenoafsa opened the pull request:
### Contexto
Rota CRUD de profiles criado com o objetivo de permitir o uso das funções POST, GET, PATCH e DELETE com os profiles. Para isso, também houve a criação dos modelos: ProfileBase, Profile, ProfileCreate e ProfilePatch. Também houve a refatoração das rotas de tanto posts quanto profiles, ambas agora estão na pasta routes e são chamadas na main.py

### Descrição
- Rotas CRUD profile:
   - POST /profiles/: Rota para criar um novo profile;
   - GET /profiles/{ong_id}: Rota para listar todos os profiles de uma ONG específica;
   - PATCH /profiles/{ong_id}/{id}: Rota para atualizar um profile existente;
   - DELETE /profiles/{ong_id}/{id}: Rota para deletar um profile existente.
- Modelos profile:
   - ProfileBase: Modelo base usado para herdar características como name e email para os demais modelos;
   - ProfileCreate: Modelo de entrada usado para a criação de um profile;
   - Profile: Modelo par avalidação;
   - PostPatch: Modelo usado para atualização de um profile.

Commit:
feat

added profiles CRUD and refacted how the CRUDS are called in main.py

pedroStelzer reviewed (approved):
Muito bom, Breno! Vi teu código e parece estar tudo correto. Tanto a implementação do CRUD de perfil quanto a refatoração do código.

mateusriff commented on models.py:
```diff
@@ -1,23 +1,32 @@
+    ong: int = Field(nullable=False)
+
+class ProfileCreate(ProfileBase):
+    ong: int = Field(nullable=False)
```
O modelo de perfil (profile) deveria conter todos os dados de ONGs que são retornados pela API do Bora Impactar. Mais detalhes foram fornecidos pessoalmente.

Commit:
fix/refactionCrudPosts

added all the requirements for an ONG

mateusriff commented on models.py:
```diff
@@ -1,32 +1,100 @@
+    logo_url: Optional[str] = Field(default=None)
+    profile_id: int = Field(foreign_key="profile.id")
+    
+    profile: "Profile" = Relationship(back_populates="sustainable_development_goals")
```
Passar `table=true` [significa que você quer que o modelo criado seja uma tabela na base de dados](https://fastapi.tiangolo.com/tutorial/sql-databases/#create-models), e não queremos criar tabelas para `GalleryImage`, `Skill`, `Cause` e `SDG`. Sugiro manter esses 4 modelos, mas remover `table=true` deles. Já que não vão ser tabelas, relações também devem ser removidas deles, como essa abaixo:
```py
profile: "Profile" = Relationship(back_populates="causes")
```
Então, por exemplo, `Skill` deveria ficar dessa forma:
```py
class Skill(SQLModel):
    id: int | None = Field(default=None, primary_key=True)
    name: str = Field()
    profile_id: int = Field(foreign_key="profile.id")  # na realidade, não sei se deveria ter esse profile_id, confere se Skill tem isso naquela mensagem que Kiev mandou. Se não tiver (acho que não tem), pode apagar.
```

mateusriff commented on models.py:
```diff
@@ -1,32 +1,100 @@
+    gallery_images: List["GalleryImage"] = Relationship(back_populates="profile")
+    skills: List["Skill"] = Relationship(back_populates="profile")
+    causes: List["Cause"] = Relationship(back_populates="profile")
+    sustainable_development_goals: List["SDG"] = Relationship(back_populates="profile")
```
Já que as 4 tabelas que citei no outro comentário não vão ser tabelas da banco dados, sugiro fazer as seguintes alterações:
```suggestion
    gallery_images: Optional[List["GalleryImage"]] = Field(default=None)
    skills: Optional[List["Skill"]] = Field(default=None)
    causes: Optional[List["Cause"]] = Field(default=None)
    sustainable_development_goals: Optional[List["SDG"]] = Field(default=None)
```
Atenção: não sei se (por exemplo) `List["Skill"]` é válido se `Skill` não é uma tabela no banco de dados. Talvez sim. Talvez não. Também, tem que importar o `Optional` de `typing`. O uso de `Optional` aqui é porque eu suspeito que algumas ONGs podem não ter esses campos.

mateusriff commented on models.py:
```diff
@@ -1,32 +1,100 @@
+    x_link: Optional[str] = Field(default=None)
+    facebook_link: Optional[str] = Field(default=None)
+    pix_qr_code_link: Optional[str] = Field(default=None)
+    site: Optional[str] = Field(default=None)
```
A gente não vai permitir que dados da ONG sejam editados por meio da nossa plataforma, já que não é possível sincronizar os dados. Então esse modelo é desnecessário. Uma rota de patch também seria, por sinal.

mateusriff commented on models.py:
```diff
@@ -1,32 +1,100 @@
+# ProfileCreate model
 class ProfileCreate(ProfileBase):
-    ong: int = Field(nullable=False)
+    pass
```
O fato de `ProfileCreate` herdar de `ProfileBase` me fez pensar sobre os campos que _não_ estão em `ProfileBase`, como `skills` (que está em `Profile`). Já que deve ser possível criar um modelo passando valores para campos como `skills`, campos como `skills` deveriam estar em `ProfileBase`. Recomendo mover _todos_ os campos de `Profile` para `ProfileBase`. Deixa `Profile` assim no fim das contas:
```py
class Profile(ProfileBase, table=True):
    pass
```

Commit:
hot-fix/refactionCrudposts

fixed the PR requirements

Commit:
hot-fix/refactionCrudPosts

last changes, the other change that was asked was making the code unable to run

mateusriff committed:
bugfix: profile models

mateusriff committed:
bugfix: change default start_year field from Jesus' birth year
